### ISSUE TITLE
Handling UI configuration in a json file

### ISSUE THREAD

**Issue by objectiser** [jaegertracing/jaeger-operator#68]
Currently the CR enables options (name value pairs) to be specified which are converted to arguments supplied to the executable.

However[ UI configuration](https://www.jaegertracing.io/docs/1.7/deployment/#ui-configuration) requires the specification of a separate file.

Wondering whether all options should be defined in a configmap per jaeger instance?

**Comment by jpkrohling** [comment]
I completely missed this part. Is this config file new?

I did consider a configmap for the other parts, similar to what we have for Kubernetes/OpenShift, but I then realized that the CR *is* the configuration source. It's unclear what would be the behavior if someone changes a configmap that is managed by the operator.

In that sense, I think I would do the same here: we could have a construct similar to `Options` (or even just reuse it, I think it might work), and convert it to JSON. Like:

```yaml
apiVersion: "io.jaegertracing/v1alpha1"
kind: "Jaeger"
metadata:
  name: "ui-options"
spec:
  strategy: all-in-one
  ui:
    tracking: 
      gaID: UA-000000-2
```

**Comment by objectiser** [comment]
No its not new. Agree with your point about the configmap.

**Comment by objectiser** [comment]
We should also think about how this might link with the spark dependencies job (e.g. https://github.com/jaegertracing/jaeger-operator/issues/44) - so if the dependencies job is enabled when using production strategy, then the UI dependencies tab should also be enabled. But the converse is also true - the dependencies tab should automatically be disabled if the spark job is not enabled.

**Comment by pavolloffay** [comment]
Other related configuration is archive traces https://github.com/jaegertracing/jaeger/issues/818. IIRC there should be a button available when the archive index is specified. The c* already supports achieving.

**Comment by pavolloffay** [comment]
Here is the link for UI config file https://www.jaegertracing.io/docs/1.8/deployment/#ui-configuration

**Comment by jpkrohling** [comment]
This was addressed by #115, right?

**Comment by objectiser** [comment]
Yes
